frontend/src/components/SearchComponent.jsx

```diff
@@ -0,0 +1,31 @@
+import { Box, Button, Input } from "@mui/material";
+import SearchIcon from '@mui/icons-material/Search';
+
+const SearchComponent=()=>{
+  const handleSearch = async () => {
+    if (!map) return;
+    try {
+      const response = await fetch(`https://nominatim.openstreetmap.org/search?format=json&q=${encodeURIComponent(searchQuery)}`);
+      const data = await response.json();
+      if (data.length > 0) {
+        const { lat, lon } = data[0];
+        const center = fromLonLat([parseFloat(lon), parseFloat(lat)]);
+        setSelectedPlaceCoordinates(center);
+        map.getView().animate({ center, zoom: 10, duration: 1000 });
+      } else {
+        alert('Location not found');
```

Comment:
Given your use of material-ui, consider replacing the vanilla alert with a mui component such as Alert, Dialog, Modal, etc.